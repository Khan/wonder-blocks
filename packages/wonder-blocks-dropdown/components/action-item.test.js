import * as React from "react";
import renderer from "react-test-renderer";
import {Link} from "react-router-dom";
import {shallow} from "enzyme";
import ReactRouterEnzymeContext from "react-router-enzyme-context";

import ActionItem from "./action-item.js";

describe("ActionItem", () => {
    it("should render with disabled styles", () => {
        const actionItem = renderer.create(
            <ActionItem href="/foo" disabled={true} />,
        );

        expect(actionItem).toMatchSnapshot();
    });

    it("should render an anchor if there's no router", () => {
        const wrapper = shallow(<ActionItem href="/foo" />);

        expect(
            wrapper
                // call the props.children(state, handlers) with empty objects for both params
                .renderProp("children")({}, {})
                // shallow render the first child of that result to unwrap StyledAnchor HOC
                .first()
                .shallow()
                // check that it includes an <a> tag
                .find("a"),
        ).toHaveLength(1);
    });

    it("should render a Link if there's a router", () => {
        // Same thing as the previous test but this time include a router context
        // when shallow rendering.
        const options = new ReactRouterEnzymeContext();
        const wrapper = shallow(<ActionItem href="/foo" />, options.get());

        expect(
            wrapper
                .renderProp("children")({}, {})
                .first()
                .shallow()
                .find(Link),
        ).toHaveLength(1);
    });
});
