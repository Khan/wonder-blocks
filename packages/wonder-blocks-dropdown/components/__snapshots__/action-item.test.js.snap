// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActionItem should render with disabled styles 1`] = `
<button
  className=""
  disabled={true}
  onBlur={[Function]}
  onClick={[Function]}
  onDragStart={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchCancel={[Function]}
  onTouchEnd={[Function]}
  onTouchStart={[Function]}
  role="menuitem"
  style={
    Object {
      "::MozFocusInner": Object {
        "border": 0,
      },
      "alignItems": "center",
      "background": "#ffffff",
      "border": "none",
      "color": "rgba(33,36,44,0.32)",
      "cursor": "default",
      "display": "flex",
      "flexDirection": "row",
      "height": 40,
      "margin": 0,
      "minHeight": 40,
      "outline": "none",
      "paddingLeft": 16,
      "paddingRight": 16,
      "textDecoration": "none",
      "touchAction": "manipulation",
    }
  }
  tabIndex={-1}
  type="button"
>
  <span
    className=""
    style={
      Object {
        "MozOsxFontSmoothing": "grayscale",
        "WebkitFontSmoothing": "antialiased",
        "display": "block",
        "fontFamily": "Lato, sans-serif",
        "fontSize": 16,
        "fontWeight": 400,
        "lineHeight": "20px",
        "userSelect": "none",
        "whiteSpace": "nowrap",
      }
    }
  />
</button>
`;
